cmake/loki_transform.cmake

```diff
@@ -573,30 +582,84 @@ endfunction()
 #
 ##############################################################################
 
-function( loki_transform_ecphys )
+function( loki_transform_command )
 
-    set( options "" )
-    set( oneValueArgs MODE FRONTEND CONFIG BUILDDIR )
+    set( options CPP )
+    set( oneValueArgs COMMAND MODE FRONTEND CONFIG BUILDDIR )
     set( multiValueArgs OUTPUT DEPENDS SOURCES HEADERS )
 
     cmake_parse_arguments( _PAR "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
 
+    if( NOT _PAR_COMMAND )
+        ecbuild_critical( "No COMMAND specified for loki_transform_command" )
+    endif()
+
     set( _ARGS )
 
-    _loki_transform_parse_target_args( loki_transform_ecphys )
+    _loki_transform_parse_target_args( loki_transform_command )
     _loki_transform_env_setup()
 
-    ecbuild_debug( "COMMAND ${_LOKI_TRANSFORM} ecphys ${_ARGS}" )
+    ecbuild_debug( "COMMAND ${_LOKI_TRANSFORM} ${_PAR_COMMAND} ${_ARGS}" )
 
     add_custom_command(
         OUTPUT ${_PAR_OUTPUT}
-        COMMAND ${_LOKI_TRANSFORM} ecphys ${_ARGS}
+        COMMAND ${_LOKI_TRANSFORM} ${_PAR_COMMAND} ${_ARGS}
```

Comment:
Does this make an implicit assumption about all sub-modes of `loki-transform.py` having the same set of arguments. Is this actually true? Could I trigger C-transpilation via this from CLOUDSC? Not a showstopper, just curious, as the safety checks have now moved to the cmake layer....

Reply:
It doesn't imply a requirement on the sub-modes, only provides the opportunity to enable additional modes of `loki-transform.py` via the same CMake layer. I consider the ecphys interface to be the de-facto definition, but if we can homogenize this across more modes eventually it's certainly not a drawback. But it would definitely not work right now to call the transpilation that way.
Motivation is mostly a way for me to inject an `ecrad` mode without having to duplicate all of the CMake layer.

Reply:
With regards to arguments checking, I would prefer leaving that to the Python layer as it would otherwise require replicating that logic. So if a mode is called via the wrong CMake function, this will fail in the call to loki-transform.

Reply:
Yes, absolutely agree about the Python layer being the one to check that arguments are sane; and happy to go ahead with this PR as is. I was merely trying to point out that we need to be careful which args the cmake layer passes down, if different sub-modes accept different arguments. But more of an academic consideration at this point, so not a showstopper.